cli_tools/common/image/importer/inflater.go

```diff
@@ -227,18 +284,38 @@ func (facade *shadowTestInflaterFacade) Cancel(reason string) bool {
 	return facade.mainInflater.Cancel(reason)
 }
 
-func (facade *shadowTestInflaterFacade) compareWithShadowInflater(mainPd, shadowPd *persistentDisk, mainIi, shadowIi *shadowTestFields) string {
-	matchFormat := "sizeGb-%v,sourceGb-%v,content-%v"
+const matchFormat = "sizeGb-%v,sourceGb-%v,content-%v,qemuchecksum-%v"
+
+func (facade *shadowTestInflaterFacade) compareWithShadowInflater(mainPd, shadowPd *persistentDisk, mainIi, shadowIi *inflationInfo) string {
 	sizeGbMatch := shadowPd.sizeGb == mainPd.sizeGb
 	sourceGbMatch := shadowPd.sourceGb == mainPd.sourceGb
-	contentMatch := shadowIi.checksum == mainIi.checksum
-	match := sizeGbMatch && sourceGbMatch && contentMatch
+	contentMatch := isChecksumMatch(shadowIi.checksum, mainIi.checksum)
+	qemuChecksumMatch := "false"
+	if facade.qemuChecksum == "" {
+		qemuChecksumMatch = "skipped"
+	} else if isChecksumMatch(facade.qemuChecksum, mainIi.checksum) {
+		qemuChecksumMatch = "true"
+	}
+
+	match := sizeGbMatch && sourceGbMatch && contentMatch && (qemuChecksumMatch == "true")
 
 	var result string
 	if match {
 		result = "true"
 	} else {
-		result = fmt.Sprintf(matchFormat, sizeGbMatch, sourceGbMatch, contentMatch)
+		result = fmt.Sprintf(matchFormat, sizeGbMatch, sourceGbMatch, contentMatch, qemuChecksumMatch)
 	}
 	return result
 }
+
+func getDiskName(executionID string) string {
+	return fmt.Sprintf("disk-%v", executionID)
+}
+
+// isChecksumMatch verifies whether checksum matches, excluded useless characters.
+func isChecksumMatch(checksum1, checksum2 string) bool {
+	reg, _ := regexp.Compile("[^a-zA-Z0-9]+")
+	processedChecksum1 := reg.ReplaceAllString(checksum1, "")
+	processedChecksum2 := reg.ReplaceAllString(checksum2, "")
```

Comment:
Why is removing non alphanumeric ('useless') characters needed?

Reply:
checksum output format from shell command vs golang lib is slightly different. Comparing to operating the string to enforce the format, it's more reliable and easier to just ignore useless charactors.